backend/src/packages/chat-messages/chat-message.service.ts

```diff
@@ -49,6 +60,55 @@ class ChatMessageService implements Service {
     return chatMessage.toObject();
   }
 
+  public async generateReply(
+    payload: ChatMessageCreatePayload,
+  ): Promise<ChatMessageGetAllItemResponseDto> {
+    const sender = await this.chatbotService.getChatbotUser();
+    if (!sender) {
+      throw new UsersError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
+    }
+
+    const chatMessages = await this.chatMessageRepository.findAllByChatId(
+      payload.chatId,
+    );
+
+    const openAiMessages: OpenAiMessageGenerateRequestDto[] = chatMessages.map(
+      (chatMessage) => {
+        const { senderId, message } = chatMessage.toObject();
+
+        return {
+          role:
+            payload.senderId === senderId
+              ? OpenAiRoleKey.ASSISTANT
+              : OpenAiRoleKey.USER,
+          content: message,
+        };
+      },
+    );
+
+    openAiMessages.push({
+      role: 'user',
+      content: `Pretend that you a psychologist and you give mental support to the patient - Answer this question: '${payload.message}'. Don't mention that you are an AI model, be compassionate and short answers.`,
+    });
```

Comment:
Let's move it to constants. Add templete {{message}} and put payload.message via template helper